src/System.Net.Http/src/uap/System/Net/HttpHandlerToFilter.cs

```diff
@@ -47,9 +50,27 @@ internal HttpHandlerToFilter(RTHttpBaseProtocolFilter filter)
             }
             cancel.ThrowIfCancellationRequested();
 
-
             RTHttpRequestMessage rtRequest = await ConvertRequestAsync(request).ConfigureAwait(false);
-            RTHttpResponseMessage rtResponse = await _next.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+
+            RTHttpResponseMessage rtResponse;
+            try
+            {
+                rtResponse = await _next.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+            }
+            catch (TaskCanceledException)
+            {
+                throw;
+            }
+            catch (Exception)
```

Comment:
What if there are two separate kinds of Exception types (one saved and a completely different one here)?